src/pip/_internal/cli/req_command.py

```diff
@@ -271,6 +271,7 @@ def make_resolver(
                 force_reinstall=force_reinstall,
                 upgrade_strategy=upgrade_strategy,
                 py_version_info=py_version_info,
+                lazy_wheel='fast-deps' in options.features_enabled,
```

Comment:
Oooo! I missed `ExtrasCandidate` when I discussed this w/ @McSinyx.

I think we can cover this by changing how self.dist is populated (instead of changing how iter_dependencies works), since what's really happening in `_iter_dependencies` -- we're creating a separate distribution object and using it.

Reply:
Similar to what was done in #8448.

Reply:
@pradyunsg, I think we did discuss point (5) and what you came up with (1a28d089e03af2c86b65c560c95aadaedd117cbe) will handle it just fine.  So now we came to a consensus on how to cover the listed points above, which is different from what this PR is pushing, I'm thinking about filing another one (likely within today) to avoid intensive rebasing which I'm not exactly good at :smile:

@chrahunt, thank you for the thoughtful heads up as well as the tips on testing.